docs/advanced-features/i18n-routing.md

```diff
@@ -177,7 +177,7 @@ export function middleware(request: NextRequest) {
     request.nextUrl.locale === 'default'
 
   return shouldHandleLocale
-    ? NextResponse.redirect(`/en${request.nextUrl.href}`)
+    ? NextResponse.redirect(`/en${request.nextUrl.pathname}`)
```

Comment:
What about when there is a query string?

I think it should be something like

```js
const url = new URL(request.nextUrl)
url.pathname = `/en${url.pathname}`
NextResponse.redirect(url)
```

Reply:
Good catch @styfle! I am thinking about not to create a new object, but not sure if there is any pitfall doing this, something like
```javascript
request.nextUrl.pathname = `/en${request.nextUrl.pathname}`
NextResponse.redirect(`/en${request.nextUrl.href}`)
```